src/host/outputStream.cpp

```diff
@@ -159,12 +159,13 @@ void ConhostInternalGetSet::WarningBell()
 // Routine Description:
 // - Sets the title of the console window.
 // Arguments:
-// - title - The null-terminated string to set as the window title
+// - title - The string to set as the window title
 // Return Value:
 // - <none>
 void ConhostInternalGetSet::SetWindowTitle(std::wstring_view title)
 {
-    ServiceLocator::LocateGlobals().getConsoleInformation().SetTitle(title);
+    auto& gci = ServiceLocator::LocateGlobals().getConsoleInformation();
+    gci.SetTitle(title.empty() ? gci.GetOriginalTitle() : title);

```

Comment:
I'm not sure why the title parameter originally expected to be null-terminated, but that's no longer the case. The [`CONSOLE_INFORMATION::SetTitle`](https://github.com/microsoft/terminal/blob/ad51b22f440a57ff8381499c84e19b9f1d174cab/src/host/consoleInformation.cpp#L217) method immediately constructs a `std::wstring` with that parameter, and in the [`Terminal::SetWindowTitle`](https://github.com/microsoft/terminal/blob/ad51b22f440a57ff8381499c84e19b9f1d174cab/src/cascadia/TerminalCore/TerminalApi.cpp#L83) implementation, it's assigned to a `std::optional<std::wstring>` using `emplace`. Neither of those should need a null-terminated `string_view` as far I understand.